src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptAsScriptingOperation.cs

```diff
@@ -0,0 +1,322 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using Microsoft.SqlTools.ServiceLayer.Scripting.Contracts;
+using Microsoft.SqlTools.Utility;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Smo;
+using System.Collections.Specialized;
+using System.Text;
+using System.Globalization;
+using Microsoft.SqlServer.Management.SqlScriptPublish;
+
+namespace Microsoft.SqlTools.ServiceLayer.Scripting
+{
+    /// <summary>
+    /// Class to generate script as for one smo object
+    /// </summary>
+    public class ScriptAsScriptingOperation : SmoScriptingOperation
+    {
+        public ScriptAsScriptingOperation(ScriptingParams parameters): base(parameters)
+        {
+        }
+
+        private string BatchTerminator = "GO";
```

Comment:
Can we make this a constant in a utils file? We reuse this in a bunch of places. I hope we'll never have to change it (e.g. support non-GO operators).